src/java.base/share/classes/java/util/concurrent/LinkedTransferQueue.java

```diff
@@ -1156,8 +1157,9 @@ public void put(E e) {
      * @throws NullPointerException if the specified element is null
      */
     public boolean offer(E e, long timeout, TimeUnit unit) {
-        return offer(e);
-    }
+        Objects.requireNonNull(e);
```

Comment:
While here, it makes sense to update the timed offer variant, since it is affected in a similar way.